src/components/MainNav/AppOrderProvider.js

```diff
@@ -0,0 +1,212 @@
+ * value object to both.
+ */
+export const AppOrderContext = createContext({
+  /**
+   * whether or not the persisted preference request is in progress.
+  */
+  isLoading: true,
+  /**
+   *  Persisted array of app objects for re-ordering - the user-preferred app order.
+   *  The objects in the array have the shape:
+   *  { name: string - the module package name, sans scope and `ui-` prefix }
+   */
+  listOrder: [],
+  /**
+   * list of app link information ordered by user preference, falling back to the order from stripes-config.
+  */
+  apps: [],
+  /**
+   * Function to update the preference. Accepts an list of objects with shape:
+   * { name: string - the module package name, sans scope and `ui-` prefix }
+   */
+  updateList: () => {},
+  /**
+   * Function to delete any the app order preference and reset the list.
+  */
+  reset: () => {},
+});
+
+// hook for AppOrderContext consumption.
+export const useAppOrderContext = () => {
+  return useContext(AppOrderContext);
+};
+
+/**
+ *  returns the list of apps/app nav information as filtered by permissions.
+*/
+function getAllowedApps(appModules, stripes, pathname, lastVisited, formatMessage) {
+  const apps = appModules.map((entry) => {
+    const name = entry.module.replace(packageName.PACKAGE_SCOPE_REGEX, '');
+    const perm = `module.${name}.enabled`;
+
+    if (!stripes.hasPerm(perm)) {
+      return null;
+    }
+
+    const id = `clickable-${name}-module`;
+
+    const pathRoot = pathname.split('/')[1];
+    const entryRoot = entry.route.split('/')[1];
+    const active = pathRoot === entryRoot;
+
+    const last = lastVisited[name];
+    const home = entry.home || entry.route;
+    const href = (active || !last) ? home : lastVisited[name];
+
+    return {
+      id,
+      href,
+      active,
+      name,
+      ...entry,
+    };
+  }).filter(app => app);
+
+  /**
+   * Add Settings to apps array manually
+   * until Settings becomes a standalone app
+   */
+
+  if (stripes.hasPerm('settings.enabled')) {
+    apps.push({
+      displayName: formatMessage({ id: 'stripes-core.settings' }),
+      name: 'settings',
+      id: 'clickable-settings',
+      href: lastVisited.x_settings || SETTINGS_ROUTE,
+      active: pathname.startsWith(SETTINGS_ROUTE),
+      description: formatMessage({ id: 'stripes-core.folioSettings' }),
+      iconData: {
+        src: settingsIcon,
+        alt: formatMessage({ id: 'stripes-core.tenantSettingsLabel' }),
+        title: formatMessage({ id: 'stripes-core.settings' }),
+      },
+      route: SETTINGS_ROUTE
+    });
+  }
+  return apps;
+}
+
+/**
+ * Context provider component for AppOrderContext.
+ * @param {object} props - The component props
+ * @param {React.ReactNode} props.children - The children to render within the provider
+ * @returns {JSX.Element} - The context provider with children
+ */
+export const AppOrderProvider = ({ children }) => { // eslint-disable-line react/prop-types
```

Comment:
Why are you turning off lint here? It's courteous to leave a comment explaining what you know about this line that lint does not.